### ISSUE TITLE
Audio messages not working after update to 2.4.0

### ISSUE THREAD

**Issue by PGG106** [telegramdesktop/tdesktop#8719]
### Steps to reproduce
1. Update the telegram desktop client to 2.4.0 or any subsequent version 
2. try to send an audio message 
 

### Expected behaviour
the microphone input should be picked up
### Actual behaviour
on 2.4.0 the audio isn't sent at all
on 2.4.2 every audio is completely blank 
### Configuration
Operating system: windows 10 2004

Version of Telegram Desktop:2.4.2 (the problem is fixed with a rollback to a version older than 2.4.0)

Used theme: night mode turned on with the night theme

**Comment by PGG106** [comment]
as extra info i'll add that by going  in settings>advanced settings > call settings i can clearly see telegram is actually picking up the microphone input 
![image](https://user-images.githubusercontent.com/72303893/94991208-d22cbf00-0581-11eb-8f70-a09268b122c3.png)

**Comment by ilya-fedin** [comment]
Can you send log.txt? What is your hardware?

**Comment by PGG106** [comment]
i'm currently running the 2.3.1 version because i needed the audio messages to work, would updating and immediately sending the log still work?
as far as hardware goes i'm running this : 
![image](https://user-images.githubusercontent.com/72303893/94996174-517eba80-05a3-11eb-9a2b-90f38a9dec84.png)

**Comment by ilya-fedin** [comment]
> would updating and immediately sending the log still work?

Yes, it should

**Comment by PGG106** [comment]
i've updated any tried to record 3 sample messages (and can confirm they still don't work)
[log.txt](https://github.com/telegramdesktop/tdesktop/files/5321959/log.txt)

**Comment by ilya-fedin** [comment]
Which audio format is set to that microphone in the control panel?

**Comment by PGG106** [comment]
is this what you mean? (it is the windows default) 
![image](https://user-images.githubusercontent.com/72303893/94997145-a0c7e980-05a9-11eb-88dc-a0e6ac0cfea2.png)

**Comment by ilya-fedin** [comment]
Yeah.
1. Is `OpenAL Soft on Microfono (ROG Strix HS BLE)` the device that you wanted to record?
2. Will it start to work if you set 1 channel, 16 bit, 48000 Hz?

**Comment by PGG106** [comment]
1.the microphone in an on board microphone on a pair of asus headphones and it is indeed rog strix hs ble (i'm not sure what that openAL soft stuff means)
2. i do not have the 1 channel option in my settings menu i tried 2 channels 16 bit 48 khz but it doesn't change anything

**Comment by PGG106** [comment]
i tried every option i had in the menu
2 channels 16/24 bit 44100 and 96000 work
both the 24 bit and the 16 bit do not work on 48000

**Comment by ilya-fedin** [comment]
Open command line, type:
```
set ALSOFT_LOGLEVEL=3
set ALSOFT_LOGFILE=%CD%\openal_log.txt
```
And run Telegram in that command line by pasting the full path, openal_log.txt should appear in the working directory of command line

**Comment by PGG106** [comment]
i tried with "set ALSOFT_LOGFILE=C:\Users\giuseppe\AppData\Roaming\Telegram Desktop\openal_log.txt" but nothing appeared in the directory

**Comment by ilya-fedin** [comment]
Is it a log after recording a voice message?

**Comment by PGG106** [comment]
no i didn't think of trying to send a message after the whole cmd stuff, this one is ( i tried recording two sample messages) 
[openal_log.txt](https://github.com/telegramdesktop/tdesktop/files/5322052/openal_log.txt)

**Comment by ilya-fedin** [comment]
Try to get a log with `set ALSOFT_DRIVERS=winmm`

**Comment by PGG106** [comment]
[openal_log.txt](https://github.com/telegramdesktop/tdesktop/files/5322072/openal_log.txt)

**Comment by ilya-fedin** [comment]
> openal_log.txt

And recording didn't work anyway?

**Comment by PGG106** [comment]
no it didn't (proof attached)

![image](https://user-images.githubusercontent.com/72303893/94998276-f0f67a00-05b0-11eb-9018-52908e40d8dd.png)

**Comment by ilya-fedin** [comment]
oh, that's strange, why it doesn't record with 48000 :thinking:

**Comment by PGG106** [comment]
it's only 48khz and only from 2.4.0 onwards , this is a test i made switching from 2.4.2 to 2.3.2 without changing anything else 
![image](https://user-images.githubusercontent.com/72303893/94998319-61050000-05b1-11eb-82ad-5fbf76343198.png)

**Comment by ilya-fedin** [comment]
Yeah, I know, openal backend was changed in 2.4.0 since a lot of users complained that tehy are not able to record voice messages. The previous backend had problems with 4-channel devices and even didn't see some devices.

**Comment by ilya-fedin** [comment]
I tried to make a build with openal 1.20.1 on github: https://github.com/ilya-fedin/tdesktop/actions/runs/286877806

When the build will finish (it should take 30 min-1 hour), you can download a test binary and check if issue persists on it.

**Comment by ilya-fedin** [comment]
It is built

**Comment by PGG106** [comment]
i downloaded the new .exe and i overwrote the old one, now everytime i press on the microphone icon the audio recording istantly stops and nothing is sent

**Comment by ilya-fedin** [comment]
Can you get openal log?

**Comment by PGG106** [comment]
[log.txt](https://github.com/telegramdesktop/tdesktop/files/5322177/log.txt)
[openal_log.txt](https://github.com/telegramdesktop/tdesktop/files/5322178/openal_log.txt)

while getting the openal log the audio were actually produced and sent (they are however still blank)

**Comment by ilya-fedin** [comment]
> openal_log.txt

Looks like that was with `ALSOFT_DRIVERS=winmm`, right? Can you make a log without this variable set?

**Comment by PGG106** [comment]
yes "ALSOFT_DRIVERS=winmm" was turned on, without it the audio messages do not even turn up blank (they don't get created at all) here's a log without it
[openal_log.txt](https://github.com/telegramdesktop/tdesktop/files/5322184/openal_log.txt)
[log.txt](https://github.com/telegramdesktop/tdesktop/files/5322185/log.txt)

**Comment by ilya-fedin** [comment]
I guess it is better to fill a bugreport to https://github.com/kcat/openal-soft/issues with the openal log, I'm out of ideas, this is something low-level :(

**Comment by PGG106** [comment]
will do,if you think the issue can be closed go on and do it

**Comment by ilya-fedin** [comment]
I think it is better to leave this issue open until this would be fixed somewhere. I will write a comment around your issue in openal tracker if there would be something to clarify for openal devs.

**Comment by PGG106** [comment]
i have one last question, should i revert to the last official binary or keep using the one you asked me to try? because with the official one i can just change the 48kh to something else and it works, but with this one i can't send anything at all

**Comment by ilya-fedin** [comment]
test binary is built with debug versions of libraries and really slower than regular one + it uses test api keys, it is not good to use it more than is needed for tests

shorter answer: use release binary until test binary is needed for tests

**Comment by ilya-fedin** [comment]
Should be fixed with 2.4.4

**Comment by PGG106** [comment]
yes i can confirm it now works ,i'll close the issue
